aws/resource_aws_dx_lag.go

```diff
@@ -0,0 +1,162 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDxLag() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDxLagCreate,
+		Read:   resourceAwsDxLagRead,
+		Update: resourceAwsDxLagUpdate,
+		Delete: resourceAwsDxLagDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"band_width": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDxLagBandWidth,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"num_connections": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"force_destroy": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceAwsDxLagCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.CreateLagInput{
+		ConnectionsBandwidth: aws.String(d.Get("band_width").(string)),
+		LagName:              aws.String(d.Get("name").(string)),
+		Location:             aws.String(d.Get("location").(string)),
+		NumberOfConnections:  aws.Int64(int64(d.Get("num_connections").(int))),
+	}
+	resp, err := conn.CreateLag(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.LagId)
+	return resourceAwsDxLagRead(d, meta)
+}
+
+func resourceAwsDxLagRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	lagId := d.Id()
+	input := &directconnect.DescribeLagsInput{
+		LagId: aws.String(lagId),
+	}
+	resp, err := conn.DescribeLags(input)
+	if err != nil {
+		d.SetId("")
+		return err
+	}
+	if len(resp.Lags) != 1 {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] Number of DX Lag (%s) isn't one, got %d", lagId, len(resp.Lags))
+	}
+	if d.Id() != *resp.Lags[0].LagId {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] DX Lag (%s) not found", lagId)
```

Comment:
It makes sense to error out if wrong ID was found (not sure how/why would that happen, but strange things do happen sometimes), but I'm not sure if it's the reason for removing LAG from state.